specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3637,181 +3966,496 @@
+        },
+        "storageBlobUri": {
+          "type": "string",
+          "description": "The uri to the storage blob which contains the VHD. Image type must be StorageBlob."
+        },
+        "customId": {
+          "type": "string",
+          "description": "The resource id of the custom image or shared image. Image type must be custom."
+        }
+      }
+    },
+    "MarketPlaceInfoProperties": {
+      "description": "Image configurations of HostPool.",
+      "type": "object",
+      "properties": {
+        "offer": {
+          "type": "string",
+          "description": "The offer of the image."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "The publisher of the image."
+        },
+        "sku": {
+          "type": "string",
+          "description": "The sku of the image."
+        },
+        "exactVersion": {
+          "type": "string",
+          "description": "The exact version of the image."
+        }
+      }
+    },
+    "DomainInfoProperties": {
+      "description": "Domain configurations of HostPool.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The domain a virtual machine connected to a hostpool will join."
+        },
+        "joinType": {
+          "enum": [
+            "ActiveDirectory",
+            "AzureActiveDirectory"
+          ],
+          "x-ms-enum": {
+            "name": "DomainJoinType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "ActiveDirectory",
+                "description": "Using microsoft active directory."
+              },
+              {
+                "value": "AzureActiveDirectory",
+                "description": "Using microsoft azure active directory."
+              }
+            ]
+          },
+          "type": "string",
+          "description": "The type of domain join done by the virtual machine.",
+          "x-nullable": false
+        },
+        "mdmProviderGuid": {
+          "type": "string",
+          "description": "The MDM Provider GUID used during MDM enrollment for Azure AD joined virtual machines."
+        },
+        "credentials": {
+          "$ref": "#/definitions/CredentialsProperties",
+          "description": "Credentials needed to create the virtual machine."
+        }
+      }
+    },
+    "CredentialsProperties": {
+      "description": "Credentials needed to create the virtual machine.",
+      "type": "object",
+      "properties": {
+        "localAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The local admin credentials."
+        },
+        "domainAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The domain admin credentials."
+        }
+      }
+    },
+    "KeyvaultCredentialProperties": {
```

Comment:
key vault is 2 words, therefore it should be "KeyVaultCredentialProperties"? btw I found some properties start with capital and some are not, would you please help me to understand?